### PR TITLE
Charlesmchen/sae filenames 

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

This reverts most of https://github.com/signalapp/Signal-iOS/pull/3017

I realized that we can get the best of both worlds by only using the "coerce item load to data" if none of the item provider's UTI types conform to URL.  Most shares will have at least one UTI type that does, and if it does we should be able to load the share as a URL instead of as data which is desirable.

### CODE REVIEW

File: SignalShareExtension/ShareViewController.swift [hunk 1]
```diff
@@ -690,12 +695,28 @@ public class ShareViewController: UIViewController, ShareViewDelegate, SAEFailed
...
+            // as a URL:
+            //
+            // * So that we can preserve the source filename which we infer from the URL.
+            // * So that we can avoid loading the item into memory if possible.
+
+            itemProvider.loadItem(forTypeIdentifier: srcUtiType, options: nil, completionHandler: loadCompletion)
+        } else {
+            // Otherwise, fall back to using the "load as data" coercion hack since this is
+            // safer.
+            itemProvider.loadData(forTypeIdentifier: srcUtiType, options: nil, completionHandler: loadCompletion)
```
Comment (michaelkirk) [inline SignalShareExtension/ShareViewController.swift]: Do you have a test case/file for this? I'm unclear when we execute this code.

**Comment by charlesmchen** [comment]
@michaelkirk I just found an edge case where "coercing to load as data" doesn't work, so I'm going to close this PR.
